modules/distributor/distributor.go

```diff
@@ -526,20 +561,52 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 	return keys, traces, nil
 }
 
-func recordDiscaredSpans(err error, userID string, spanCount int) {
-	s := status.Convert(err)
-	if s == nil {
-		return
-	}
-	desc := s.Message()
+func countDiscaredSpans(responses []*tempopb.PushResponse, traces []*rebatchedTrace, repFactor int) (int, int, int) {
+	maxLiveDiscardedCount := 0
+	traceTooLargeDiscardedCount := 0
+	internalErrorDiscardedCount := 0
```

Comment:
i don't think we need to count internal error here. `DoBatch()` will do the work of accounting for an error while sending to the ingesters.

i think we can drop the error handling/response == nil logic here and simplify.